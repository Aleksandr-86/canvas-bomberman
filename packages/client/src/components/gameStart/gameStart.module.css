.gameStart {
  display: flex;
  flex-direction: column;
  gap: 50px;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  color: white;
  font-size: 50px;
  line-height: 1.2em;
}

.gameStartButton {
  border-color: #fff;
  background-color: transparent;
  color: #fff;
}
