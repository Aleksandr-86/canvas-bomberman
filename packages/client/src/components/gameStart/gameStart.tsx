import React from 'react'
import { Button } from '../button/button'
import styles from './gameStart.module.css'
import keyboard from '../../assets/images/keyboard.png'

interface Props {
  startGame: () => void
}

export const GameStart: React.FC<Props> = ({ startGame }) => {
  return (
    <div className={styles.gameStart}>
      <p className={styles.gameStartText}>Куда жать, чтобы было весело?!</p>
      <img
        src={keyboard}
        width={720}
        height={303}
        alt="Клавиатура"
        draggable={false}
      />
      <Button className={styles.gameStartButton} onClick={startGame}>
        ПОНЯЛ, ПОГНАЛИ!
      </Button>
    </div>
  )
}
