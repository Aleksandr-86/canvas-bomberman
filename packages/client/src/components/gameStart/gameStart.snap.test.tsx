import * as renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { GameStart } from './gameStart'
import configureStore from 'redux-mock-store'

const mockStore = configureStore([])
const store = mockStore({})

describe(`<GameStart/> snapshot test cases`, () => {
  test(`<GameStart/> should render <div className=gameStart> parent with child elements`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <GameStart />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
