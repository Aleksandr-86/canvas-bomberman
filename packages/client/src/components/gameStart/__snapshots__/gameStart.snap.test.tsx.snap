// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameStart/> snapshot test cases <GameStart/> should render <div className=gameStart> parent with child elements 1`] = `
<div
  className="gameStart"
>
  <p>
    Куда жать, чтобы было весело?!
  </p>
  <img
    alt="Клавиатура"
    draggable={false}
    height={303}
    width={720}
  />
  <button
    className="button gameStartButton"
    onClick={[Function]}
  >
    ПОНЯЛ, ПОГНАЛИ!
  </button>
</div>
`;
